backends/guest/common/verify.c

```diff
@@ -95,13 +96,15 @@ static int get_current_esl_data(const char *esl_file, uint8_t **current_esl_data
 		return INVALID_FILE;
 
 	buffer = get_data_from_file(esl_file, SIZE_MAX, &buffer_size);
-	if (buffer != NULL) {
+	if (buffer != NULL && buffer_size >= TIMESTAMP_LEN) {
 		if (buffer_size == DEFAULT_PK_LEN) {
 			print_raw(buffer, buffer_size);
```

Comment:
I don't know, I assume this is guest specific magic. I'll add a verbose check here though since regardless of *whatever* this is doing, it probably shouldn't just print stuff.

Reply:
Correct me if I am wrong, but since we are extracting the esl_data from the variable path, buffer_size will be greater than TIMESTAMP_LEN because it's a valid esl_file and the first 16 bytes are reserved for timestamp? So this check is not required?

Reply:
Generally yes, the current variable path is should only contain valid data but `secvarctl` also allows the user to pass in a custom path to treat as the current list of variables (this is useful for testing). At the end of the day, the check prevents possible buffer overread. Since we could be reading data provided by the user, I would prefer keeping this conditional